### PR TITLE
Expose info for each TSDB

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit exposes the label set alongside the min and max time for each TSDB covered by a Store.

This information is used to scope the min time for a remote query so that we do not produce partial aggregates in distriuted mode.

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: pkg/store/proxy.go [hunk 1]
```diff
@@ -254,6 +262,14 @@ func (s *ProxyStore) TimeRange() (int64, int64) {
 	return minTime, maxTime
 }
 
+func (s *ProxyStore) TSDBInfos() []infopb.TSDBInfo {
+	infos := make([]infopb.TSDBInfo, 0)
+	for _, store := range s.stores() {
+		infos = append(infos, store.TSDBInfos()...)
```
Comment (GiedriusS) [inline pkg/store/proxy.go:268]: Do we need to filter out `nil` here if a store doesn't support this call?

**Review by GiedriusS — DISMISSED** [review]
LGTM, just one comment

File: pkg/store/proxy.go (hunk 1 above)
Comment (fpetkovski) [reply pkg/store/proxy.go:268]: it seems that the spread operator can already handle nils. I added a test as a safeguard to make sure we don't have some weird regression in the future.

**Review by saswatamcode — APPROVED** [review]
LGTM!

**Pull request merged**